File: Lab1_sem2/Lab1_sem2/Function.cpp
#include "Function.h"


void edit_file(string path)
{
	ofstream text;
	text.open(path, ios::app);
	if (!text)
	{
		cout << "Error";
	}
	else
	{
		string line;
		text_out(path);
		cout << "Write the new strings. To finish enter ctrl+z\n";
		cin.ignore();
		while (getline(cin, line, '\26'))
		{
			text << line;
		}
	}
	text.close();
}

void create_file(string path)
{
	ofstream text(path);
	if (!text)
	{
		cout << "Error";
	}
	else
	{
		string line;
		cout << "Write the text. To finish enter ctrl+z" << endl;
		cin.ignore();
		while (getline(cin, line, '\26'))
		{
			text << line;
		}
	}
	text.close();
}

void text_out(string path)
{
	ifstream text(path);
	if (!text)
	{
		cout << "Error";
	}
	else
	{
		string line;
		while (!text.eof())
		{
			getline(text, line);
			cout << line << endl;
		}
	}
	text.close();
}

void text_sort(string path, vector<string>& str)
{
	ifstream text(path);
	if (!text)
	{
		cout << "Error";
	}
	else
	{
		string line;
		while (!text.eof())
		{
			getline(text, line);
			str.push_back(line);
		}

		empty_delete(str);

		string temp = "";
		for (int i = 0; i < str.size() - 1; i++)
		{
			for (int j = 0; j < str.size() - 1; j++)
			{
				if (str[j].length() - count(str[j].begin(), str[j].end(), ' ') > str[j + 1].length() - count(str[j+1].begin(), str[j+1].end(), ' '))
				{
					temp = str[j];
					str[j] = str[j + 1];
					str[j + 1] = temp;
				}
			}
		}
	}
	text.close();
}

void empty_delete(vector<string>& str)
{
	for (int i = 0; i < str.size(); i++)
	{
		if (str[i].length() - count(str[i].begin(), str[i].end(), ' ') == 0)
		{
			str.erase(str.begin() + i);
		}
	}
}

void count_sym(string path, string pathNew, vector<string> str)
{
	ofstream newFile(pathNew);
	string line = "";
	for (int i = 0; i < str.size(); i++)
	{
		newFile << str[i] << " - " << to_string(str[i].size() - count(str[i].begin(), str[i].end(), ' ')) << '\n';
	}
	newFile.close();
}

File: Lab1_sem2/labPY/Function.py
def count_sym(fileStr, pathNew):
    file = open(pathNew, 'w')
    for i in range(len(fileStr)):
        line = fileStr[i] + ' - ' + str(len(fileStr[i]) - fileStr[i].count(' ')) + '\n'
        file.write(line)
    file.close()

def edit_file(path):
    file = open(path, 'a')
    text_out(path)
    print("Write the new strings. To finish enter ctrl+x")
    while(1):
        line = input()
        if(ord(line[0]) == 24):
            break
        file.write(line + "\n")
    file.close()

def text_sort(path, fileStr):
    file = open(path, 'r')
    fileStr = file.read().split('\n')
    fileStr = list(filter(None, fileStr))

    temp = ""
    for i in range(len(fileStr) - 1):
        for j in range(len(fileStr) - 1):
            if (len(fileStr[j]) - fileStr[j].count(' ') > len(fileStr[j+1]) - fileStr[j+1].count(' ')):
                temp = fileStr[j]
                fileStr[j] = fileStr[j+1]
                fileStr[j+1] = temp

    file.close()
    return fileStr

def create_file(path):
    file = open(path, 'w')
    print("Write the text. To finish enter ctrl+x")
    while(1):
        line = input()
        if(ord(line[0]) == 24):
            break
        file.write(line + "\n")
    file.close()

def text_out(path):
    file = open(path, 'r')
    for line in file:
        print(line)
    file.close()
